projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+
+Let's look at an example action of initiating a login request.
+
+<code-example header="login-page.actions.ts">
+import { Action } from '@ngrx/store';
+
+export class Login implements Action {
+  readonly type = '[Login Page] Login';
+
+  constructor(public payload: { username: string; password: string }) {}
+}
+</code-example>
+As you can see above we implemented the `Action` interface by creating the class `Login`. In doing so we also used the `readonly` keyword on the `type` field. Thereby we ensured that you can't change the value of `type` after we've instantiated an object from the `Login` class. 
+
+We did another thing as well, namely adding the `payload` to the constructor of the `Login` class and we used the keyword `public` to do it. Using an accessor like `public`, `private` or `protected` in a constructor is a nice short hand for creating the field as well as pass it in as a parameter into the constructor. The below code in plain Javascript looks like the following:
+
+<code-example header="Action in plain JavaScript">
+class Login {
+  constructor(payload) {
+    this.type = '[Login Page] Login';
+    this.payload = payload;
+  }
+}
+</code-example>
+As you can see above we lose quite a bit of features when using plain JavaScript like: 
+- having fields like `payload` created for us via the constructor, 
+- ensuring we can't change the `type` value after instantiation. 
+- we would also need to perform checks on the `payload` being passed in to ensure it contains a `username` and `password`. 
+
+At this point I hope you want to return to the nice and typesafe world of Typescript. Lets talk about Action creation next. 
+
+### Creating an Action
+Ok, time has come to instantiate an object from our `Login` class and thereby create an action. One of the common places to create an action is in the context of being inside of a component and responding to a user event. So imagine we are inside of `login-page.component.ts` and the user has just entered their `username` and `password` and clicked on a button to login. At this point the `clicked()` method is invoked on our component. The code in the component looks as follows:
+
+<code-example header="login-page.component.ts">
+click(username: string, password: string) {
+  store.dispatch(new Login({ username: username, password: password }));
+}
+</code-example>
+We do two things here:
+- instantiate an action and pass in an object literal containing our username and password
+- dispatch the action by calling `store.dispatch()`
+
+When we instantiate our action `Login` we see that we pass an object literal. Wasn't that what we were trying to get away from by using Typescript? Actually the compiler helps us here because we declare that the constructor will accept a parameter `payload` that requires a `username` and `password` as properties. Let's show that part again from the definition of our `Login` action:
```

Comment:
```suggestion
When we instantiate our action `Login` we see that we pass an object literal. Wasn't that what we were trying to get away from by using TypeScript? Actually the compiler helps us here because we declare that the constructor will accept a parameter `payload` that requires a `username` and `password` as properties. Let's show that part again from the definition of our `Login` action:
```